eth-custodian/contracts/EthCustodianProxy.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+import '@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import {EthCustodian} from './EthCustodian.sol';
+import {SelectivePausableUpgradable} from './SelectivePausableUpgradable.sol';
+
+contract EthCustodianProxy is
+    Initializable,
+    UUPSUpgradeable,
+    AccessControlUpgradeable,
+    SelectivePausableUpgradable
+{
+    bytes32 public constant PAUSABLE_ADMIN_ROLE = keccak256('PAUSABLE_ADMIN_ROLE');
+
+    uint constant UNPAUSED_ALL = 0;
+    uint constant PAUSED_DEPOSIT_TO_EVM = 1 << 0;
+    uint constant PAUSED_DEPOSIT_TO_NEAR = 1 << 1;
+    uint constant PAUSED_WITHDRAW = 1 << 2;
+    uint constant PAUSED_WITHDRAW_PRE_MIGRATION = 1 << 3;
+
+    error AlreadyMigrated();
+    error ProducerAccountIdTooLong(bytes newProducerAccount);
+    error ProofFromPostMergeBlock();
+
+    bytes public preMigrationProducerAccount;
+    uint64 public migrationBlockHeight;
+
+    EthCustodian public ethCustodianImpl;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        EthCustodian _ethCustodianImpl
+    ) public initializer {
+        ethCustodianImpl = _ethCustodianImpl;
+        __Pausable_init();
+        __AccessControl_init();
+        __UUPSUpgradeable_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _grantRole(PAUSABLE_ADMIN_ROLE, _msgSender());
+    }
+
+    function depositToNear(
+        string memory nearRecipientAccountId
+    ) external payable whenNotPaused(PAUSED_DEPOSIT_TO_NEAR) {
```

Comment:
It's a valid point. Updating the front-end would require a few more changes even though they are small. Are we okay with that @karim-en ?

Reply:
I think it is ok to break the backward compatibility here, even I propose to remove the `depositToEvm` because it is already deprecated and isn't used by the web-app